src/containers/auth/resetPassword/index.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { PureComponent } from 'react';
+import styles from './styles.module.scss';
+import { resetApprove, resetPassword } from '../../../services/auth';
+import Spinner from '../../../components/spinner/index';
+import validateAuth from '../../../validation/auth';
+
+class ResetPassword extends PureComponent {
+    state = {
+        success: false,
+        status: null,
+        password: null,
+        passwordConfirm: null,
+        linkId: null,
+        message: null
+    }
+
+    componentDidMount() {
+        const params = new URLSearchParams(location.search);
+        this.sendApprove(params.get('id'));
+    }
+
+    sendApprove = async (linkId) => {
+        const { status, data } = await resetApprove(linkId);
+        if (data.name === 'Error') {
+            this.setState({
+                status: 400,
+                linkId: null,
+            });
+        } else {
+            this.setState({
+                status,
+                linkId
+            });
+        }
+    }
+
+    sendReset = async (e) => {
+        e.preventDefault();
+        const { linkId, password, passwordConfirm } = this.state;
+        const errors = validateAuth({ password });
+        if (password && passwordConfirm) {
+            if (password === passwordConfirm) {
+                if (errors.password) {
+                    this.setState({
+                        message: errors.password
+                    });
+                } else {
+                    const { status, data } = await resetPassword({ linkId, password, passwordConfirm });
+                    if (data.status === 200) {
```

Comment:
You should rewrite this 4 `if`. It looks awful

Reply:
Yep. I agree with you.

Reply:
rewrited